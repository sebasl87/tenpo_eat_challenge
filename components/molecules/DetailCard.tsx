import React from 'react'
import styled from 'styled-components/native';
import { View } from 'react-native';

export interface IDetailCard {
    discount?: string;
    restaurantName?: string;
    rate?: string;
    lapseTime?: string;

}

const ImageStar = styled.Image`
    width: 16px;
    height: 16px;
    margin-right: 2px;
`

const RestaurantNameContainer = styled.View`
width: 100%;
max-width: 104px;
align-items: center;
display: flex;
`

const RestaurantRate = styled.View`
flex-direction: row;
display: flex;
align-items: center;
`

const ContainerRate = styled.View`
flex-direction: row;
width: 100%;
max-width: 104px;
display: flex;
justify-content: space-between;
align-items: center;
`

const NameRestaurant = styled.Text`
color: #333333;
font-size: 14px;
`

const Rate = styled.Text`
color: #333333;
font-size: 12px;
`

const LapseTimeText = styled.Text`
color: #333333;
font-size: 12px;
`

const DetailCard: React.FC<IDetailCard> = ({ restaurantName, rate, lapseTime }) => {
    return (
        <View>
            <RestaurantNameContainer><NameRestaurant style={{ fontFamily: "Gotham-Medium" }}>{restaurantName}</NameRestaurant></RestaurantNameContainer>
            <ContainerRate>
                <RestaurantRate>
                    <ImageStar
                        source={require("../../assets/imgs/starmini.png")}
                    />
                    <Rate style={{ fontFamily: "Gotham-Light" }}>{rate}</Rate>
                </RestaurantRate>
                <View>
                    <LapseTimeText style={{ fontFamily: "Gotham-Light" }}>{lapseTime} min</LapseTimeText>
                </View>
            </ContainerRate>
        </View>
    );
}


export default DetailCard;