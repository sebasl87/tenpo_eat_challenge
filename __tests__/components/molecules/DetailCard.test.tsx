import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import DetailCard from "../../../src/components/molecules/DetailCard";


describe("DetailCard component", () => {
    it("should render successfully with DetailCard", () => {
        const titleSection = renderer
            .create(<DetailCard rate="DetailCard_mock" lapseTime="DetailCard_mock" />)
            .toJSON();

        expect(titleSection).toMatchSnapshot();
    });
});
