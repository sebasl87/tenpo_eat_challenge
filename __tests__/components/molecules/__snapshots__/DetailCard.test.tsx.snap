// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCard component should render successfully with DetailCard 1`] = `
<View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "maxWidth": 104,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#333333",
            "fontSize": 14,
          },
          {
            "fontFamily": "Gotham-Medium",
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "maxWidth": 104,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/imgs/starmini.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "marginRight": 2,
              "width": 16,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#333333",
              "fontSize": 12,
            },
            {
              "fontFamily": "Gotham-Medium",
            },
          ]
        }
      >
        DetailCard_mock
      </Text>
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "#333333",
              "fontSize": 12,
            },
            {
              "fontFamily": "Gotham-Medium",
            },
          ]
        }
      >
        DetailCard_mock
         min
      </Text>
    </View>
  </View>
</View>
`;
